openmls/src/group/mls_group/create_commit.rs

```diff
@@ -48,40 +49,59 @@ impl MlsGroup {
 
         let sender_index = self.sender_index();
         // Make a copy of the current tree to apply proposals safely
-        let mut provisional_tree = RatchetTree::new_from_public_tree(&self.tree());
+        let mut diff: TreeSyncDiff = self.tree().empty_diff()?;
 
         // Apply proposals to tree
         let apply_proposals_values =
-            match provisional_tree.apply_proposals(backend, proposal_queue, &[]) {
-                Ok(res) => res,
-                Err(_) => return Err(CreateCommitError::OwnKeyNotFound.into()),
-            };
+            self.apply_proposals(&mut diff, backend, proposal_queue, &[])?;
```

Comment:
This doesn't have to be part of this PR, but I think it's worthwhile mentioning it now:
We discussed that it would make sense to return a `StagedCommit` as well as the actual commit message. I'm wondering to what extent that would change the strategy here already.

Reply:
I filed #617 for that.